azplugins/flow.py

```diff
@@ -746,13 +767,51 @@ def density(self):
             return np.zeros(self.bins)
 
     @property
-    def velocity(self):
-        r"""The current average velocity profile."""
+    def mass_density(self):
+        r"""The current mass-averaged density profile."""
+        if hoomd.comm.get_rank() != 0:
+            hoomd.context.msg.error('Flow profile only defined on root rank.\n')
+            raise RuntimeError('Flow profile only defined on root rank')
+
+        if self.samples > 0:
+            return np.divide(self._bin_mass, self._counts, out=np.zeros(self.bins), where=self._counts > 0)
```

Comment:
With the rewrite I did above, I think you just divide by the volume and samples, like for the number density,